### PR TITLE
Allow gem to be loaded even if the VDDK is not found

### PR DESCRIPTION
**Pull request by Fryguy**
- Add FFI::VixDiskLib::API.available? method to allow for checking
whether or not the VDDK is available.

@agrare Please review.

### CODE REVIEW

**Comment by Fryguy** [comment]
This is a follow-on from https://github.com/ManageIQ/manageiq-providers-vmware/pull/501#discussion_r362531495 .  I this this will allow us to verify whether the VDDK is available or not, properly.  However, I'm not sure if the old behavior of it blowing up on load was expected or not.

File: lib/ffi-vix_disk_lib/api.rb [hunk 1]
```diff
@@ -5,44 +5,43 @@ module VixDiskLib
     module API
       extend FFI::Library
 
-      def attach_function(*args)
-        super
+      def self.attach_function(*args)
```
Comment (Fryguy) [inline lib/ffi-vix_disk_lib/api.rb]: Seems this hasn't been working for a long time :(

**Comment by agrare** [comment]
LGTM but I can't merge in this repo

File: lib/ffi-vix_disk_lib/api.rb [hunk 2]
```diff
@@ -5,44 +5,43 @@ module VixDiskLib
...
+          false
         end
       end
+      @load_error = "ffi-vix_disk_lib: failed to load any version of VixDiskLib!" if loaded_library.nil?
 
-      unless @load_error || loaded_library.length > 0
-        STDERR.puts load_errors.join("\n")
-        @load_error = "ffi-vix_disk_lib: failed to load any version of VixDiskLib!"
-      end
       LOADED_LIBRARY = loaded_library
```
Comment (carbonin) [inline lib/ffi-vix_disk_lib/api.rb]: I think this changed from `""` to `nil` in the case that we don't load something. Is that a problem?

Comment (Fryguy) [reply lib/ffi-vix_disk_lib/api.rb]: I don't think this value is actually used anywhere.

**Comment by Fryguy** [comment]
I'd like to get some actual manual tests against this, but I'm not sure the best way.

**Comment by Fryguy** [comment]
After discussion with @agrare, we're going to let this blow up when the library is not available, as we are concerned there are code paths that otherwise won't account for this...I'll make a separate PR to clean up *how* it blows up.

**Pull request closed without merging**